src/jit/codegenarmarch.cpp

```diff
@@ -730,20 +741,52 @@ void CodeGen::genPutArgStk(GenTreePutArgStk* treeNode)
                 structOffset += (2 * TARGET_POINTER_SIZE);
                 nextIndex += 2;
             }
+#else  // _TARGET_ARM_
+            // For a >= 4 byte structSize we will generate a ldr and str instruction each loop
+            //             ldr     r2, [r0]
+            //             str     r2, [sp, #16]
```

Comment:
How about `str r2, [sp, #4]` ?
Because we are going to increase offset by `TARGET_POINTER_SIZE` for ARM32 here.